src/Messages.jl

```diff
@@ -455,25 +459,20 @@ function decode(m::Message, encoding::String="gzip")::Vector{UInt8}
 end
 
 # Writing HTTP Messages to IO streams
-"""
-    HTTP.httpversion(::Message)
-
-e.g. `"HTTP/1.1"`
-"""
-httpversion(m::Message) = "HTTP/$(m.version.major).$(m.version.minor)"
+Base.write(io::IO, v::HTTPVersion) = write(io, "HTTP/", string(v.major), ".", string(v.minor))
 
 """
     writestartline(::IO, ::Message)
 
 e.g. `"GET /path HTTP/1.1\\r\\n"` or `"HTTP/1.1 200 OK\\r\\n"`
 """
 function writestartline(io::IO, r::Request)
-    write(io, "$(r.method) $(r.target) $(httpversion(r))\r\n")
+    write(io, r.method, " ", r.target, " ", r.version, "\r\n")
     return
 end
 
 function writestartline(io::IO, r::Response)
-    write(io, "$(httpversion(r)) $(r.status) $(statustext(r.status))\r\n")
+    write(io, r.version, " ", string(r.status), " ", statustext(r.status), "\r\n")
```

Comment:
I think this is causing a test failure; the problem is that we're dispatching to this method in Base:
```julia
function write(io::IO, x1, xs...)
    written::Int = write(io, x1)
    for x in xs
        written += write(io, x)
    end
    return written
end
```
which isn't ideal when writing to a Socket since it might actually "flush" the writes out on the wire at any time, but it means the response might be received in fragments. This becomes worse when you have a system under heavy load and we have a scenario where it writes `write(socket, "HTTP/1.1")`, then gets task switched away for several minutes before switching back and writing `write(socket, 200)`.

But what's the best way to solve while trying to avoid allocations? Not super clear, but one idea is that we have a `writebuffer::IOBuffer` in the `Connection` object where these writes would go and then we make sure we have a dedicated `flush` operation at some point (like after `writestartline`, `writeheaders`, `writebody`, etc.).